core/src/main/java/org/infinispan/interceptors/locking/OptimisticLockingInterceptor.java

```diff
@@ -39,68 +30,27 @@
  */
 public class OptimisticLockingInterceptor extends AbstractTxLockingInterceptor {
 
-   private LockAcquisitionVisitor lockAcquisitionVisitor;
-   private boolean needToMarkReads;
-
    private static final Log log = LogFactory.getLog(OptimisticLockingInterceptor.class);
-
-   @Override
-   protected Log getLog() {
-      return log;
-   }
+   private boolean needToMarkReads;
 
    @Start
    public void start() {
-      if (cacheConfiguration.clustering().cacheMode() == CacheMode.LOCAL &&
+      needToMarkReads = cacheConfiguration.clustering().cacheMode() == CacheMode.LOCAL &&
```

Comment:
@pruivo Off-topic, do you have any idea why we only mark reads in LOCAL mode? The `unreliableReturnValue()` check also seems wrong, because it shouldn't affect `cache.get()`.